lib/routes/yystv/fetcher.ts

```diff
@@ -0,0 +1,34 @@
+import { DataItem } from '@/types';
+import cache from '@/utils/cache';
+import ofetch from '@/utils/ofetch';
+
+export async function fetchArticleContentCached(link: string): Promise<string> {
+    const result = await cache.tryGet(link, () => ofetch(link));
+
+    if (typeof result === 'string') {
+        return result;
+    }
+
+    return '';
+}
+
+export async function fetchDataItemCached(link: string, processor: (articleContent: string) => DataItem): Promise<DataItem> {
+    const dataItemKey = `dataitem:${link}`;
+    const result = await cache.tryGet(dataItemKey, async () => {
+        const result = await fetchArticleContentCached(link);
+        const processedItem = processor(result);
+
+        return processedItem;
+    });
+
+    if (typeof result !== 'object') {
+        const articleContent = await fetchArticleContentCached(link);
+        const processedItem = processor(articleContent);
+
+        cache.set(dataItemKey, processedItem);
+
+        return processedItem;
+    }
+
+    return result as DataItem;
+}
```

Comment:
These will cache the article twice, one with whole HTML document and another one with DataItem properties. Only cache it once with DataItem properties which is what the current code https://github.com/DIYgod/RSSHub/blob/92312865411c3286e5cf603663251031937db87f/lib/routes/yystv/docs.ts#L54-L59 does.